native/core/src/execution/jni_api.rs

```diff
@@ -287,7 +287,7 @@ fn prepare_output(
     let results = output_batch.columns();
     let num_rows = output_batch.num_rows();
 
-    if results.len() != num_cols {
+    if results.len() < num_cols {
```

Comment:
Why can we now get more output columns (i.e., `results.len() > num_cols`) than before?

Reply:
A failed query has no output on the scan. I.e., the query asks for empty column. But the native scan still outputs all columns. It should be more reasonable to output no column, actually. Let me see if it could be fixed at the native scan.

Reply:
I changed this back after pushing empty projection to the native scan.